frontend/catalyst/utils/toml.py

```diff
@@ -52,25 +55,65 @@ def read_toml_file(toml_file: str) -> TOMLDocument:
     return config
 
 
-def check_mid_circuit_measurement_flag(config: TOMLDocument) -> bool:
-    """Check the global mid-circuit measurement flag"""
-    return bool(config.get("compilation", {}).get("mid_circuit_measurement", False))
+@dataclass(unsafe_hash=True)
```

Comment:
What does unsafe hash do?

Reply:
Unsafe hash relaxes Python's built-in `__hash__` precautions. Here the dataclass contains lists which are mutable structures, but logically it is immutable, thus the flag. See [official docs](https://docs.python.org/3/library/dataclasses.html#module-contents).